test/lseq.rs

```diff
@@ -0,0 +1,110 @@
+
+use crdts::lseq::*;
+use crdts::lseq::ident::*;
+use rand::distributions::Alphanumeric;
+use rand::Rng;
+
+#[test]
+fn test_out_of_order_inserts() {
+    let mut site1 = LSeq::new(0);
+    site1.local_insert(0, 'a');
+    site1.local_insert(1, 'c');
+    site1.local_insert(1, 'b');
+
+    assert_eq!(site1.text(), "abc");
+}
+
+#[test]
+fn test_inserts() {
+    // A simple smoke test to ensure that insertions work properly.
+    // Uses two sites which random insert a character and then immediately insert it into the
+    // other site.
+    let mut rng = rand::thread_rng();
+
+    let mut s1 = rng.sample_iter(Alphanumeric);
+    let mut s2 = rng.sample_iter(Alphanumeric);
+
+    let mut site1 = LSeq::new(0);
+    let mut site2 = LSeq::new(1);
+
+    for _ in 0..5000 {
+        if rng.gen() {
+            let op = site1.local_insert(rng.gen_range(0, site1.text().len() + 1), s1.next().unwrap());
+            site2.apply(&op);
+        } else {
+            let op = site2.local_insert(rng.gen_range(0, site2.text().len() + 1), s2.next().unwrap());
+            site1.apply(&op);
+        }
+    }
+    assert_eq!(site1.text(), site2.text());
+}
+
+#[derive(Clone)]
+struct OperationList(pub Vec<Op>);
+
+use quickcheck::{Arbitrary, Gen};
+
+impl Arbitrary for OperationList {
```

Comment:
this is an attempt at generating complex series of operations but in practice i'm not convinced it tests enough edgecases.